01_Data/src/layers/sequelize/model/Tariff/Tariffs.ts

```diff
@@ -1,32 +1,295 @@
+import {Price} from "./Price";
+import {TariffElement, TariffElementData} from "./TariffElement";
+
+export const ALL_TARIFF_TYPES = [
+  'AD_HOC_PAYMENT',
+  'PROFILE_CHEAP',
+  'PROFILE_FAST',
+  'PROFILE_GREEN',
+  'REGULAR'
+] as const;
+
+export type TariffType = typeof ALL_TARIFF_TYPES[number];
+
+function isTariffType(value: string): value is TariffType {
+  return (ALL_TARIFF_TYPES as readonly string[]).includes(value);
+}
+
+export function tariffType(value: string): TariffType {
+  assert(isTariffType(value), `Invalid tariff type: ${value}`);
+  return value;
+}
 
 @Table
-export class Tariff extends Model {
+export class Tariff extends Model implements TariffData {
+
   static readonly MODEL_NAME: string = Namespace.Tariff;
 
-  /**
-   * Fields
-   */
+  public static newInstance(data: TariffData): Tariff {
+    return Tariff.build({...data}, {include: [TariffElement]});
+  }
+
+  @PrimaryKey
+  @Column({
+    type: DataType.STRING(36),
+    validate: {
+      isTariffId: (value: string) => assert(isTariffId(value), `Invalid id: ${value}`),
+    }
+  })
+  declare id: TariffId;
 
   @Index
+  @Column(DataType.STRING)
+  declare stationId: string; // TODO: ??
+
+  @PrimaryKey
   @Column({
-    type: DataType.STRING,
-    unique: 'stationId_unit',
+    type: DataType.CHAR(2),
+    allowNull: false,
+    validate: {
+      isCountryCode: (value: string) => assert(isCountryCode(value), `Invalid countryCode: ${value}`),
+    }
+  })
+  declare countryCode: CountryCode;
+
+  @PrimaryKey
+  @Column({
+    type: DataType.STRING(3),
+    allowNull: false,
+    validate: {
+      isPartyId: (value: string) => assert(isPartyId(value), `Invalid partyId: ${value}`),
+    }
+  })
+  declare partyId: PartyId;
+
+  @Column({
+    type: DataType.CHAR(3),
+    allowNull: false,
+    validate: {
+      isCurrency: (value: string) => assert(isCurrency(value), `Invalid currency: ${value}`),
+    }
   })
-  declare stationId: string;
+  declare currency: Currency;
 
   @Column({
     type: DataType.STRING,
-    unique: 'stationId_unit',
+    validate: {
+      isTariffType: (value: string) => assert(isTariffType(value), `Invalid type: ${value}`),
+    }
+  })
+  declare type?: TariffType;
+
+  @Column({
+    type: DataType.JSON
+  })
+  declare tariffAltText?: Displaytext[];
+
+  @Column({
+    type: DataType.STRING(255),
+    validate: {
+      url: (value: string) => assert(isUrl(value), `Invalid tariffAltUrl: ${value}`),
+    }
+  })
+  declare tariffAltUrl?: Url;
+
+  @Column({
+    type: DataType.DECIMAL(19, 4),
+    get(this: Tariff) {
+      return parseFloat(this.getDataValue('minPriceExclVat'));
+    }
+  })
+  private declare minPriceExclVat?: number;
+
+  @Column({
+    type: DataType.DECIMAL(19, 4),
+    get(this: Tariff) {
+      return parseFloat(this.getDataValue('minPriceInclVat'));
+    }
+  })
+  private declare minPriceInclVat?: number;
+
+  @Column(DataType.VIRTUAL)
```

Comment:
Similarly to restrictions for TariffElement, I believe it's better to maintain min price and max price as separate columns in the Tariff table.